src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -323,14 +341,27 @@ class decklink_producer : public IDeckLinkInputCallback
                       const spl::shared_ptr<core::frame_factory>& frame_factory,
                       const std::string&                          vfilter,
                       const std::string&                          afilter,
+                      const std::wstring&                         format,
                       bool                                        freeze_on_lost)
         : device_index_(device_index)
         , format_desc_(format_desc)
         , frame_factory_(frame_factory)
         , freeze_on_lost_(freeze_on_lost)
-        , video_filter_(vfilter, AVMEDIA_TYPE_VIDEO, format_desc_, mode_)
-        , audio_filter_(afilter, AVMEDIA_TYPE_AUDIO, format_desc_, mode_)
+        , input_format(format_desc_)
+        , vfilter_(vfilter)
+        , afilter_(afilter)
     {
+        // use user-provided format if available, or choose the channel's output format
+        if (!format.empty()) {
+            input_format = core::video_format_desc(format);
+        }
+
+        mode_         = get_display_mode(input_, input_format.format, bmdFormat8BitBGRA, bmdVideoOutputFlagDefault);
+        video_filter_ = Filter(vfilter_, AVMEDIA_TYPE_VIDEO, format_desc_, mode_);
+        audio_filter_ = Filter(afilter_, AVMEDIA_TYPE_AUDIO, format_desc_, mode_);
+
+        field_count_ = mode_->GetFieldDominance() != bmdProgressiveFrame && mode_->GetFieldDominance() != bmdProgressiveSegmentedFrame ? 2 : 1;
```

Comment:
I don't think we want to do this check any more. The only thing we use field_count_ for is to set the size of the frame_buffer, so the field could be removed.

As the format could change after initialisation we should prepare the buffer for that and do a `frame_buffer_.set_capacity(3);` below instead